ruby/Ice/filesystem/client.rb

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env ruby
+# Copyright (c) ZeroC, Inc.
+
+require 'etc'
+require 'Ice'
+
+# Load the code generated by the Slice compiler. The Slice module Filesystem maps to a Ruby module with the same name.
+require_relative 'Filesystem.rb'
+
+# Recursively print the contents of directory "dir" in tree fashion while showing the contents of each file.
+def listRecursive(dir, depth)
+    indent = ''
+    depth = depth + 1
+    for i in (0...depth)
+        indent += "\t"
+    end
+ 
+    contents = dir.list()
+ 
+    for node in contents
+        subdir = Filesystem::DirectoryPrx::checkedCast(node)
+        file = Filesystem::FilePrx::uncheckedCast(node)
+        print indent + node.name()
+        if subdir
+            puts "(directory):"
+            listRecursive(subdir, depth)
+        else
+            puts "(file):"
+            text = file.read()
+            for line in text
+                puts indent + "\t" + line
+            end
+        end
+    end
+end
+
+# Create an Ice communicator. We'll use this communicator to create proxies and manage outgoing connections.
+# It also accepts additional command line arguments.
+Ice::initialize(ARGV) do |communicator|
+    # Create a proxy for the root directory.
```

Comment:
We should create proxies with the new preferred API for creating proxies, like in all other demos.